contracts/src/unruggable_memecoin.cairo

```diff
@@ -161,7 +169,21 @@ mod UnruggableMemecoin {
         }
 
         fn transfer(ref self: ContractState, recipient: ContractAddress, amount: u256) -> bool {
+
             let sender = get_caller_address();
+
+            if self.launched.read() == false {
+                // enforce max number of holders before launch
+                let holders_count = self.pre_launch_holders_count.read();
+                assert(
+                    holders_count < MAX_HOLDERS_BEFORE_LAUNCH, 
```

Comment:
```suggestion
                    holders_count < MAX_HOLDERS_BEFORE_LAUNCH, 
```

if we want at most 10 holders, we should use `<=`
if we want to gas-optimize, keep `<` but set `MAX_HOLDERS_BEFORE_LAUNCH` to 11

Reply:
create an error module with constants, see #18

Reply:
>if we want at most 10 holders, we should use <=
if we want to gas-optimize, keep < but set MAX_HOLDERS_BEFORE_LAUNCH to 11

`holders_count` there means the number of holders before the transfer function is called. so at that point, we are asserting that the current number of holders is less than the max to ensure that there is a free slot which the new recipient can use

Reply:
ah yeah ok! makes sense